src/register.py

```diff
@@ -97,48 +88,53 @@ def write_int(self, uintvalue, blindwrite=False, word_offset=0):
         self.parent.write_int(device_name=self.name, integer=uintvalue, blindwrite=blindwrite, word_offset=word_offset)
 
     def _write_common(self, **kwargs):
-        # TODO - make read before write optional - otherwise just write the whole word based on last-known values
         """
-        :param kwargs:
+        Form the dictionary of values that must be written
+        :param kwargs: the field names and values to write
         :return:
         """
-        if len(kwargs) == 0:
+        if len(kwargs.keys()) == 0:
```

Comment:
Should be no need to make this change -- len() on a dict should work like you want it to.